### PR TITLE
Make error messages match for infix, unary op, identifiers

### PR DESCRIPTION
**Pull request by prachi1210**
Partial work for https://github.com/taskcluster/json-e/issues/168

### CODE REVIEW

File: jsone/interpreter.py [hunk 1]
```diff
@@ -22,7 +22,8 @@
 
 
 def expectationError(operator, expected):
```
Comment (djmitche) [inline jsone/interpreter.py]: this should probably be renamed to `infixExpectationError` since it's specific to infix operators now.

Comment (prachi1210) [reply jsone/interpreter.py]: This function is also called while throwing expectation error for unary operators. Should I still rename it?

Comment (djmitche) [reply jsone/interpreter.py]: https://github.com/prachi1210/json-e/blob/1d29d55bffd4967ba61b2e6472abbfc080586ba6/jsone/interpreter.py#L80-L92 is raising InterpreterErrors directly.

Comment (djmitche) [reply jsone/interpreter.py]: @prachi1210 do you want to finish this up?

**Comment by prachi1210** [comment]
Yes. @djmitche I was taken sick the past week. Will get to it asap!

**Comment by djmitche** [comment]
@prachi1210 I added that rename and I'll get this merged now once it passes tests.

**Pull request merged**